ccl/kvccl/kvfollowerreadsccl: TestSecondaryTenantFollowerReadsRouting failed

ccl/kvccl/kvfollowerreadsccl.TestSecondaryTenantFollowerReadsRouting [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12447077?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12447077?buildTab=artifacts#/) on master @ [bfc5c35b112fe612d837ef247b2c61e532e8184d](https://github.com/cockroachdb/cockroach/commits/bfc5c35b112fe612d837ef247b2c61e532e8184d):


```
=== RUN   TestSecondaryTenantFollowerReadsRouting
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/becd5e00c91ac217b69adea284b6c9ea/logTestSecondaryTenantFollowerReadsRouting1802064653
    test_log_scope.go:81: use -show-logs to present logs inline
    followerreads_test.go:1115: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/becd5e00c91ac217b69adea284b6c9ea/logTestSecondaryTenantFollowerReadsRouting1802064653
--- FAIL: TestSecondaryTenantFollowerReadsRouting (291.73s)
=== RUN   TestSecondaryTenantFollowerReadsRouting/shared-process
    followerreads_test.go:1088: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/followerreads_test.go:1088
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestSecondaryTenantFollowerReadsRouting/shared-process
    --- FAIL: TestSecondaryTenantFollowerReadsRouting/shared-process (233.21s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSecondaryTenantFollowerReadsRouting.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>



Jira issue: CRDB-33045

ccl/kvccl/kvfollowerreadsccl.TestSecondaryTenantFollowerReadsRouting [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12512301?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12512301?buildTab=artifacts#/) on master @ [90a1f850ccfc9a123c9dcaba317296151ae32a8c](https://github.com/cockroachdb/cockroach/commits/90a1f850ccfc9a123c9dcaba317296151ae32a8c):


```
=== RUN   TestSecondaryTenantFollowerReadsRouting
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/becd5e00c91ac217b69adea284b6c9ea/logTestSecondaryTenantFollowerReadsRouting1798776363
    test_log_scope.go:81: use -show-logs to present logs inline
    followerreads_test.go:1115: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/becd5e00c91ac217b69adea284b6c9ea/logTestSecondaryTenantFollowerReadsRouting1798776363
--- FAIL: TestSecondaryTenantFollowerReadsRouting (260.07s)
=== RUN   TestSecondaryTenantFollowerReadsRouting/shared-process
    followerreads_test.go:1088: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/followerreads_test.go:1088
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestSecondaryTenantFollowerReadsRouting/shared-process
    --- FAIL: TestSecondaryTenantFollowerReadsRouting/shared-process (211.98s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113733 ccl/kvccl/kvfollowerreadsccl: TestSecondaryTenantFollowerReadsRouting failed [C-test-failure O-robot T-kv branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSecondaryTenantFollowerReadsRouting.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvfollowerreadsccl.TestSecondaryTenantFollowerReadsRouting [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12543751?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12543751?buildTab=artifacts#/) on master @ [414cb46513079bb8869bd776f045ed6c96ad6d26](https://github.com/cockroachdb/cockroach/commits/414cb46513079bb8869bd776f045ed6c96ad6d26):


```
=== RUN   TestSecondaryTenantFollowerReadsRouting
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/becd5e00c91ac217b69adea284b6c9ea/logTestSecondaryTenantFollowerReadsRouting2549660821
    test_log_scope.go:81: use -show-logs to present logs inline
    followerreads_test.go:1115: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/becd5e00c91ac217b69adea284b6c9ea/logTestSecondaryTenantFollowerReadsRouting2549660821
--- FAIL: TestSecondaryTenantFollowerReadsRouting (201.79s)
=== RUN   TestSecondaryTenantFollowerReadsRouting/shared-process
    followerreads_test.go:1088: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/followerreads_test.go:1088
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestSecondaryTenantFollowerReadsRouting/shared-process
    --- FAIL: TestSecondaryTenantFollowerReadsRouting/shared-process (141.70s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113733 ccl/kvccl/kvfollowerreadsccl: TestSecondaryTenantFollowerReadsRouting failed [C-test-failure O-robot T-kv branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSecondaryTenantFollowerReadsRouting.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvfollowerreadsccl.TestSecondaryTenantFollowerReadsRouting [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12563825?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12563825?buildTab=artifacts#/) on master @ [cd3d2e3c7a7356c7e0604f32ff3cb4e88ed12845](https://github.com/cockroachdb/cockroach/commits/cd3d2e3c7a7356c7e0604f32ff3cb4e88ed12845):


```
=== RUN   TestSecondaryTenantFollowerReadsRouting
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/becd5e00c91ac217b69adea284b6c9ea/logTestSecondaryTenantFollowerReadsRouting558412934
    test_log_scope.go:81: use -show-logs to present logs inline
    followerreads_test.go:1115: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/becd5e00c91ac217b69adea284b6c9ea/logTestSecondaryTenantFollowerReadsRouting558412934
--- FAIL: TestSecondaryTenantFollowerReadsRouting (167.60s)
=== RUN   TestSecondaryTenantFollowerReadsRouting/shared-process
    followerreads_test.go:1088: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/followerreads_test.go:1088
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestSecondaryTenantFollowerReadsRouting/shared-process
    --- FAIL: TestSecondaryTenantFollowerReadsRouting/shared-process (111.72s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113733 ccl/kvccl/kvfollowerreadsccl: TestSecondaryTenantFollowerReadsRouting failed [C-test-failure GA-blocker O-robot T-kv branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSecondaryTenantFollowerReadsRouting.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/kvccl/kvfollowerreadsccl/followerreads_test.go
+++ b/pkg/ccl/kvccl/kvfollowerreadsccl/followerreads_test.go
@@ -46,7 +46,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/stop"
-	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing/tracingpb"
 	"github.com/cockroachdb/errors"
 	"github.com/stretchr/testify/require"
@@ -942,6 +941,7 @@ func TestSecondaryTenantFollowerReadsRouting(t *testing.T) {
 	defer utilccl.TestingEnableEnterprise()()
 
 	skip.UnderStressRace(t, "times out")
+	skip.UnderDeadlock(t)
 
 	for _, testCase := range []struct {
 		name             string
@@ -952,11 +952,6 @@ func TestSecondaryTenantFollowerReadsRouting(t *testing.T) {
 		{name: "latency-based", sharedProcess: false, validLatencyFunc: true},
 		{name: "locality-based", sharedProcess: false, validLatencyFunc: false},
 	} {
-		if syncutil.DeadlockEnabled && testCase.sharedProcess {
-			// TODO(yuzefovich): unskipping shared-process config under deadlock
-			// is tracked by #113555.
-			continue
-		}
 		t.Run(testCase.name, func(t *testing.T) {
 			const numNodes = 4
 			gatewayNode := 3

